lib/file.js

```diff
@@ -147,31 +169,74 @@ async function uploadFileConcurrently(filepath, url, options) {
                 fileSize : size
             }
         ],
-        ...options
+        ...uploadOptions
     });
 }
 
+/**
+ * @typedef {Object} UploadAEMMultipartOptions
+ *
+ * @property {String} [method] Optional HTTP method (defaults to 'PUT')
+ * @property {Number} [timeout] Optional socket timeout
+ * @property {Object} [headers] Optional override of request headers
+ * @property {Number} [retryMaxDuration=60000] time to retry until throwing an error (ms)
+ * @property {Number} [retryMaxCount=5] time to retry until throwing an error (ms)
+ * @property {Number} [retryInterval=100] time between retries, used by exponential backoff (ms)
+ * @property {Boolean} [retryEnabled=true] retry on failure enabled
+ * @property {Boolean} [retryAllErrors=false] whether or not to retry on all http error codes or just >=500
+ * @property {Number} [preferredPartSize] Optional custom preferred part size. Might be adjusted depending on the target.
+ * @property {Number} [maxConcurrent] Optional concurrent amount.
+ */
+/**
+ * @typedef {Object} UploadAEMMultipartTarget
+ *
+ * @property {String[]} urls URLs
+ * @property {Number} [maxPartSize] Maximum size of each part
+ */
 /**
  * Upload multi part file using multiple simultaneous transfers
  * Throws the first unrecoverable error if unsuccessful, all others are logged
- * @param {Array.<String|URL>} urls Array of url locations for multipart upload (e.g. presigned URL) 
+ * @param {UploadAEMMultipartTarget} target Target urls
  * @param {String} filepath Path to locally saved file
- * @param {UploadFileOptions} options Additional options such as headers, retry features, etc 
+ * @param {UploadAEMMultipartOptions} options Additional options such as headers, retry features, etc 
  * @returns {Promise} resolves when upload completes
  */
-async function uploadMultiPartFileConcurrently(filepath, urls, options) {
-    // options contains header information
+async function uploadMultiPartFileConcurrently(filepath, target, options={}) {
+    // fail fast, keep error handling the same as older, non-concurrent function
+    if (!target) {
+        throw new IllegalArgumentError('target not provided', target);
+    } else if (!target.urls || target.urls.length === 0) {
+        throw new IllegalArgumentError('\'target.urls\' but be a non-empty array', target.urls);
+    } else if (!isPositiveNumber(target.maxPartSize)) {
+        throw new IllegalArgumentError('\'target.maxPartSize\' must be a positive number', target.maxPartSize);
+    }
+
+    // Calculate the partSize based on the number of urls
     const { size } = await getFileStats(filepath);
+
+    // to stay backwards compatible with older `uploadAEMMultipartFile` that uses `partSize`
+    options.preferredPartSize = options.partSize? options.partSize: options.preferredPartSize;
```

Comment:
`blockUpload` uses `preferredPartSize` instead of `partSize`.  i think it is a better more descriptive name than `partSize`: https://github.com/adobe/node-httptransfer/blob/9357ef6ed96e263cbefe3cd894cdd7b59abf5f0c/lib/aemmultipart.js#L27
so i am fine to keep it. However, to keep bakwards compatibility, I added this

Reply:
additionally, very strange but `partSize` is in the `options` and `maxPartSize` is in the `target` object: https://github.com/adobe/node-httptransfer/blob/9357ef6ed96e263cbefe3cd894cdd7b59abf5f0c/lib/aemmultipart.js#L37
I just kept the behavior the same.